runtime/lua/vim/shared.lua

```diff
@@ -651,18 +651,21 @@ function vim.tbl_islist(t)
     return false
   end
 
-  local num_elem = vim.tbl_count(t)
-
-  if num_elem == 0 then
+  if next(t) == nil then
     return getmetatable(t) ~= vim._empty_dict_mt
-  else
-    for i = 1, num_elem do
-      if t[i] == nil then
-        return false
-      end
+  end
+
+  local j = 1
+  for i in
+    pairs(t--[[@as table<any,any>]])
```

Comment:
Doesn't this rely on the fact that `pairs()` traverses table in correct order? As far as I know, It doesn't have that documented guarantee.

Reply:
Hmm, good point.

It just needs to iterate the indexes first, which I'm sure every impl does, but yes, it isn't guaranteed.

Reply:
Fixed. (I think)

Reply:
> Fixed. (I think)

Yeah, I'd say so too.